flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetTableSource.java

```diff
@@ -465,6 +479,10 @@ private String getColumnName(BinaryComparison comp) {
 			typeInfo == BasicTypeInfo.INT_TYPE_INFO) {
 			return new Tuple2<>(FilterApi.intColumn(columnName), (Integer) value);
 		} else if (typeInfo == BasicTypeInfo.LONG_TYPE_INFO) {
+			if (value instanceof Integer) {
+				Integer intValue = (Integer) value;
+				return new Tuple2<>(FilterApi.longColumn(columnName), intValue.longValue());
+			}
 			return new Tuple2<>(FilterApi.longColumn(columnName), (Long) value);
```

Comment:
You should just use `((Number) value).longValue()`.
For other types, we should modify too.

Reply:
Agree. It is more readable. Changed accordingly.